client/src/lib.rs

```diff
@@ -74,28 +92,141 @@ pub enum SubxtClientError {
+                                    {
+                                        let result = match serde_json::from_str::<Output>(
+                                            &response,
+                                        )
+                                        .expect("failed to decode request response")
+                                        {
+                                            Output::Success(success) => {
+                                                Ok(success.result)
+                                            }
+                                            Output::Failure(failure) => {
+                                                Err(JsonRpseeError::Request(
+                                                    failure.error,
+                                                ))
+                                            }
+                                        };
+
+                                        send_back
+                                            .send(result)
+                                            .expect("failed to send request response");
+                                    }
+                                }
+                            }
+
+                            FrontToBack::Subscribe {
+                                subscribe_method,
+                                params,
+                                unsubscribe_method: _,
+                                send_back,
+                            } => {
+                                let request =
+                                    Request::Single(Call::MethodCall(MethodCall {
+                                        jsonrpc: Version::V2,
+                                        method: subscribe_method,
+                                        params,
+                                        id: Id::Num(0),
+                                    }));
+
+                                let (mut send_front_sub, send_back_sub) =
+                                    mpsc::channel(DEFAULT_CHANNEL_SIZE);
+                                if let Ok(message) = serde_json::to_string(&request) {
+                                    if let Some(response) =
+                                        rpc.rpc_query(&session, &message).await
+                                    {
+                                        let result = match serde_json::from_str::<Output>(
+                                            &response,
+                                        )
+                                        .expect("failed to decode subscription response")
+                                        {
+                                            Output::Success(_) => {
+                                                Ok((
+                                                    send_back_sub,
+                                                    SubscriptionId::Num(0),
```

Comment:
the subscription id is included in the response which you could use but because it's only used when unsubscribing so it's fine to ignore/hardcode.